src/cargo/ops/lockfile.rs

```diff
@@ -43,13 +43,21 @@ pub fn write_pkg_lockfile(ws: &Workspace<'_>, resolve: &Resolve) -> CargoResult<
 
     let mut out = String::new();
 
-    // Preserve the top comments in the lockfile
-    // This is in preparation for marking it as generated
-    // https://github.com/rust-lang/cargo/issues/6180
+    // At the start of the file we notify the reader that the file is generated.
+    // Specifically Phabricator ignores files containing "@generated", so we use that.
+    out.push_str("# This file is automatically @generated by Cargo.  Not intended for manual editing.\n");
+    // and preserve any other top comments
     if let Ok(orig) = &orig {
-        for line in orig.lines().take_while(|line| line.starts_with('#')) {
-            out.push_str(line);
-            out.push_str("\n");
+        let mut comments = orig.lines().take_while(|line| line.starts_with('#'));
+        if let Some(first) = comments.next() {
+            if first != "# This file is automatically @generated by Cargo.  Not intended for manual editing." {
```

Comment:
Could this be deduplicated with the comment above? Additionally, could a line break be in here to ensure it's under 80 columns?

Reply:
> Could this be deduplicated with the comment above?

Doh, good call.

> Additionally, could a line break be in here to ensure it's under 80 columns?

I cheated and made it 1 line, under 80 columns.